### PR TITLE
udpate k8s ingress controller route policy

### LINKED ISSUE #633: IngressController can not handle Ingress rule priority correctly
**Describe the bug**
IngressController can not handle Ingress rule priority correctly

**To Reproduce**
create two Ingress in k8s like:
```yaml
...
  rules:
  - http:
      paths:
      - backend:
          service:
            name: service1
        path: /v1
        pathType: Prefix
```
and
```yaml
...
  rules:
  - host: www.megaease.com
    http:
      paths:
      - backend:
          service:
            name: service2
        path: /v1
        pathType: Prefix
```
request to URL `http://www.megaease.com/v1/hello` is forwarded to service1.

**Expected behavior**
the request is forwarded to service2

**Version**
v1.5.2

### PR DESCRIPTION
**Pull request by suchen-sci**
fix #633

### CODE REVIEW

File: pkg/object/ingresscontroller/translator.go [hunk 1]
```diff
@@ -366,11 +370,23 @@ func (st *specTranslator) translateIngressRules(b *httpServerSpecBuilder, ingres
...
-			b.Rules = append(b.Rules, r)
+			if len(rule.Host) > 0 {
+				if rule.Host[0] == '*' {
+					wildcardHostRules = append(wildcardHostRules, r)
+				} else {
+					hostRules = append(hostRules, r)
+				}
+			} else {
+				noHostRules = append(noHostRules, r)
+			}
```
Comment (localvar) [inline pkg/object/ingresscontroller/translator.go:381]: the new rules have not been added to `b.Rules` in this case, that's the `existingRule` may have a wrong value.

File: pkg/object/ingresscontroller/translator.go [hunk 2]
```diff
@@ -317,6 +317,10 @@ func (st *specTranslator) translateTLSConfig(b *httpServerSpecBuilder, ingresses
 }
 
 func (st *specTranslator) translateIngressRules(b *httpServerSpecBuilder, ingress *apinetv1.Ingress) {
+	hostRules := []*httpserver.Rule{}
+	wildcardHostRules := []*httpserver.Rule{}
+	noHostRules := []*httpserver.Rule{}
```
Comment (localvar) [inline pkg/object/ingresscontroller/translator.go:322]: there could be more than one ingress, we need to consider this case also.

Comment (suchen-sci) [reply pkg/object/ingresscontroller/translator.go]: updated!

**Review by localvar — APPROVED** [review]

**Review by zhao-kun — APPROVED** [review]
LGTM

**Pull request merged**